pkg/doc/signature/ed25519signature2018/suite.go

```diff
@@ -36,6 +42,89 @@ type signer interface {
 	Sign(data []byte) ([]byte, error)
 }
 
+// CryptoSigner defines a signer which uses crypto instance to sing the message using particular key handle.
+type CryptoSigner struct {
+	cr crypto.Crypto
+	kh interface{}
+}
+
+// NewCryptoSigner creates a new CryptoSigner.
+func NewCryptoSigner(cr crypto.Crypto, kh interface{}) *CryptoSigner {
+	return &CryptoSigner{
+		cr: cr,
+		kh: kh,
+	}
+}
+
+// Sign will sign document and return signature
+func (s CryptoSigner) Sign(msg []byte) ([]byte, error) {
+	return s.cr.Sign(msg, s.kh)
+}
+
+type verifier interface {
+	// Verify will verify a signature.
+	Verify(pubKeyValue interface{}, doc, signature []byte) error
+}
+
+// LocalCryptoVerifier defines a verifier which is based on Local KMS and Crypto
+// which uses keyset.Handle as input for verification.
+type LocalCryptoVerifier struct {
+	cr       crypto.Crypto
+	localKMS *localkms.LocalKMS
+}
+
+// NewLocalCryptoVerifier creates a new LocalCryptoVerifier.
+func NewLocalCryptoVerifier(cr crypto.Crypto, localKMS *localkms.LocalKMS) *LocalCryptoVerifier {
+	return &LocalCryptoVerifier{
+		cr:       cr,
+		localKMS: localKMS,
+	}
+}
+
+// Verify will verify a signature.
+func (v *LocalCryptoVerifier) Verify(pubKeyValue interface{}, doc, signature []byte) error {
+	var keyHandle *keyset.Handle
+
+	switch pubKey := pubKeyValue.(type) {
+	case *keyset.Handle:
+		keyHandle = pubKey
+	default:
+		// TODO support *sigverifier.PublicKey case after a fix of
+		//  https://github.com/hyperledger/aries-framework-go/issues/1483
+		return errors.New("bad key handle format")
+	}
+
+	return v.cr.Verify(signature, doc, keyHandle)
+}
+
+// PublicKeyVerifier verifies a Ed25519 signature taking Ed25519 public key bytes as input.
+type PublicKeyVerifier struct {
+}
+
+// Verify will verify a signature.
+func (v *PublicKeyVerifier) Verify(pubKeyValue interface{}, doc, signature []byte) error {
+	pubKey, ok := pubKeyValue.(*sigverifier.PublicKey)
+	if !ok {
+		return errors.New("invalid public key")
+	}
+
+	if pubKey.Type != kms.Ed25519Type {
+		return fmt.Errorf("invalid public key type: %s", pubKey.Type)
+	}
+
+	// ed25519 panics if key size is wrong
+	if len(pubKey.Value) != ed25519.PublicKeySize {
+		return errors.New("ed25519: bad public key length")
+	}
+
+	verified := ed25519.Verify(pubKey.Value, doc, signature)
```

Comment:
this should be part of the kms/crypto implementation @Baha-sk

Reply:
I think @kdimak wants to separate signature verification using raw key []byte from Tink keys and this verify uses []byte keys with crypto.ed25519 package.

So `PublicKeyVerifier` is for keys with raw bytes and `LocalKeyVerifier` is for Tink keys (with public key loaded from a DID doc as []byte and transformed by localkKMS into *keyset.Handle which requires a fix I'm currently working on)

Reply:
This package should not know such details (that would be the reason for an empty interface).